pkg/cloud/services/networking/floatingip.go

```diff
@@ -22,24 +22,61 @@ import (
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
 	infrav1 "sigs.k8s.io/cluster-api-provider-openstack/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-openstack/pkg/record"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 )
 
-func (s *Service) CreateFloatingIPIfNecessary(openStackCluster *infrav1.OpenStackCluster, ip string) error {
-	fp, err := checkIfFloatingIPExists(s.client, ip)
+func (s *Service) ReconcileControlPlaneEndpoint(openStackCluster *infrav1.OpenStackCluster) error {
+
+	if openStackCluster.Spec.ControlPlaneEndpoint.Host != "" && openStackCluster.Spec.ControlPlaneEndpoint.Port != 0 {
```

Comment:
what should we do if one of them is set? Is this a valid case?

Reply:
Both will be set. If one of them is not set, validation error occurs.